src/send-reminders.js

```diff
@@ -0,0 +1,178 @@
+    ctpClient = _ctpClient
+    logger = _logger
+
+    const activeStateId = await _fetchActiveStateId()
+
+    const orderQuery =
+      await _buildQueryOfTemplateOrdersThatIsReadyToSendReminder(activeStateId)
+
+    for await (const templateOrders of ctpClient.fetchPagesGraphQl(orderQuery))
+      await pMap(
+        templateOrders,
+        (order) => _processTemplateOrder(activeStateId, order),
+        { concurrency: 3 }
+      )
+  } catch (err) {
+    logger.error(
+      'Failed on send reminders, processing should be restarted on the next run.' +
+        `Error: ${JSON.stringify(serializeError(err))}`
+    )
+  }
+
+  return stats
+}
+
+async function _fetchActiveStateId() {
+  const {
+    body: { id },
+  } = await apiRoot.states().withKey({ key: 'Active' }).get().execute()
+  return id
+}
+
+async function _buildQueryOfTemplateOrdersThatIsReadyToSendReminder(
+  activeStateId
+) {
+  const now = new Date().toISOString()
+  const where = `state(id="${activeStateId}") AND custom(fields(nextReminderDate <= "${now}"))`
+  return {
+    queryBody: `query TemplateOrdersThatIsReadyToSendReminderOrdersQuery($limit: Int, $where: String) {
+            orders (limit: $limit, where: $where) {
+                results {
+                    id
+                    orderNumber
+                    version
+                }
+            }
+        }`,
+    endpoint: 'orders',
+    variables: {
+      limit: 100,
+      where,
+    },
+  }
+}
+
+async function _processTemplateOrder(
+  activeStateId,
+  { id, orderNumber, version }
+) {
+  let retryCount = 0
+  const maxRetry = 10
+  stats.processedTemplateOrders++
+  // eslint-disable-next-line no-constant-condition
+  while (true)
+    try {
+      await apiRoot
+        .orders()
+        .withId({ ID: id })
+        .post({
+          body: {
+            actions: [
+              {
+                action: 'transitionState',
+                state: {
+                  typeId: 'state',
+                  key: 'SendReminder',
+                },
+              },
+            ],
+            version,
+          },
+        })
+        .execute()
+      stats.updatedTemplateOrders++
+      break
+    } catch (err) {
+      if (err.statusCode === 409) {
+        retryCount += 1
+        const currentVersion = await _fetchCurrentVersionOnRetry(
+          id,
+          activeStateId
+        )
+        if (!currentVersion) {
+          stats.skippedTemplateOrders++
+          break
+        }
+
+        if (retryCount > maxRetry) {
+          const retryMessage =
+            'Got a concurrent modification error when updating state from (Active to SendReminder)' +
+            ` of the template order with number "${orderNumber}".` +
+            ` Version tried "${version}",` +
+            ` currentVersion: "${currentVersion}".`
+          throw new VError(
+            err,
+            `${retryMessage} Won't retry again` +
+              ` because of a reached limit ${maxRetry}` +
```

Comment:
`Won't retry again since maximum retry limit of ${maxRetry} is reached.`

Reply:
https://github.com/commercetools/commercetools-subscriptions/pull/8/commits/5ab114875dc7e294154e6a70b67da8a1fe7929e8